sklearn_numba_dpex/kmeans/tests/test_kmeans.py

```diff
@@ -277,3 +283,52 @@ def test_select_samples_far_from_centroid_kernel(dtype):
     assert (
         selected_samples_idx[n_selected_gt_threshold:-n_selected_eq_threshold] == 100
     ).all()
+
+
+@pytest.mark.parametrize("dtype", float_dtype_params)
+def test_kmeans_plusplus_output(dtype):
+    """Test adapted from sklearn's test_kmeans_plusplus_output"""
+    # Check for the correct number of seeds and all positive values
+    data = sklearn_test_data.astype(dtype)
+
+    centers, indices = driver.kmeans_plusplus(
+        X=data, sample_weight=None, n_clusters=sklearn_test_n_clusters, random_state=42
+    )
+
+    # Check there are the correct number of indices and that all indices are
+    # positive and within the number of samples
+    assert indices.shape[0] == sklearn_test_n_clusters
+    assert (indices >= 0).all()
+    assert (indices <= data.shape[0]).all()
+
+    # Check for the correct number of seeds and that they are bound by the data
```

Comment:
We can test that it's either float32 or float64, but I think that more than that is a bit more that what we are set for, since the dtype can change depending on the device...

Reply:
OK, then for now, I am for not asserting dtype invariance and for not casting arrays to other dtype if not strictly required.

Reply:
I'm having trouble understanding what you're refering too with so much negatives :grin: do you suggest to revert the change I just made to this test, or change it, or change the behavior of the estimator, or none of those ?